### DISCUSSION TITLE
Introducing a captcha solver in SearXNG

### DISCUSSION THREAD

**Discussion by unixfox** [searxng/searxng#2458] [Ideas]
In the light #2011, it seems like some engines are serving a CAPTCHA if they deem that the SearXNG is doing too many requests. I have been thinking that we could introduce the ability to automatically solve CAPTCHA using external services for the compatible engines.

The issue with these services is that they employ workers in Third World countries. But some do not, like this service: https://capmonster.cloud/. Ideally, we should then only implement services that doesn't involve humans.

Pinging @return42 what do you think?

**Comment by return42** [comment]
I have not yet formed a comprehensive opinion on this.  I know that the massive requests on the servers come from bots (on my server 99% of the requests are from bots that have endless IPs).  Before we solve the CAPTCHAs for these bots I would rather put more resources into the defense of the bots, which is why I am currently working on a modification of the limiter to open it for new, complementary defense methods.
